realm/realm-library/src/objectServer/java/io/realm/SyncSession.java

```diff
@@ -580,7 +581,12 @@ protected void onError(AuthenticateResponse response) {
                 onGoingAccessTokenQuery.set(false);
                 RealmLog.debug("Session[%s]: Failed to get access token (%s)", configuration.getPath(),
                         response.getError().getErrorCode());
-                if (!isClosed && !Thread.currentThread().isInterrupted()) {
+                if (!isClosed
+                        && !Thread.currentThread().isInterrupted()
+                        // We might be interrupted while negotiating an access token with the Realm Object Server
+                        // This will result in a InterruptedIOException from OkHttp. We should ignore this as
+                        // well.
```

Comment:
when OkHTTP throw `InterruptedIOException`, will it retry later? Or the `InterruptedIOException` thrown because of we called `cancel()`?

Reply:
I'm not sure checking `InterruptedIOException` is necessary 

`InterruptedIOException` extends `IOException`  which will be captured in
https://github.com/realm/realm-java/blob/21503edb7a93288db49f1180104c0ff91c17d5fa/realm/realm-library/src/objectServer/java/io/realm/internal/network/AuthenticateResponse.java#L52-L55

as `IO_EXCEPTION` so `onError` should not be called since retry will be called in https://github.com/realm/realm-java/blob/a137a819f99635962363fd481caabb2509a5a020/realm/realm-library/src/objectServer/java/io/realm/internal/network/ExponentialBackoffTask.java#L37-L42

https://github.com/realm/realm-java/blob/a137a819f99635962363fd481caabb2509a5a020/realm/realm-library/src/objectServer/java/io/realm/internal/network/ExponentialBackoffTask.java#L72-L74